### PR TITLE
Add guard to prevent more than one instance of bitcore

### PR DESCRIPTION
**Pull request by eordano**

### CODE REVIEW

**Comment by eordano** [comment]
See:
* https://github.com/bitpay/bitcore/issues/1164
* https://github.com/bitpay/bitcore-message/issues/15

File: index.js [hunk 1]
```diff
@@ -1,5 +1,14 @@
 var bitcore = module.exports;
 
+var inBrowser = typeof process === 'undefined' || typeof process.versions === 'undefined';
+if ((inBrowser && window._bitcore) || (!inBrowser && global._bitcore)) {
+  throw new Error('More than one instance of bitcore found');
+}
+if (inBrowser) {
+  window._bitcore = true;
```
Comment (maraoz) [inline index.js]: I'd save the bitcore version, so that the error can be more detailed

Comment (eordano) [inline index.js]: Addressed

**Comment by fanatid** [comment]
But how this solve the problem? As I see, if I import `bitcore-mnemonic` first and then trying importing `bitcore` -- error will be raised.. I'm wrong?

**Comment by eordano** [comment]
@fanatid we'll release an updated version of `bitcore-mnemonic` with this patch: https://github.com/bitpay/bitcore-message/commit/a328b3491744d38416614971beeee495e8ffb18f

**Comment by maraoz** [comment]
ACK

**Comment by olalonde** [comment]
This won't raise any error if bitcore is required first though right? (e.g.: `require ('bitcore');require('bitcore-message')`.

**Comment by fanatid** [comment]
@olalonde as I understand, with new version of bitcore and other submodules, we should get
```bash
$ npm ls | grep bitcore
├─┬ bitcore@a.b.c
├── bitcore-mnemonic@a.b.c
```
instead
```bash
$ npm ls | grep bitcore
├─┬ bitcore@a.b.c
├─┬ bitcore-mnemonic@a.b.c1
│ └─┬ bitcore@a.b.c2
```

in first case only one bitcore will be used, so everything (I hope) should be fine, but we must be careful with versions, otherwise error will be raised

**Pull request merged**